### PR TITLE
adding sqldatabase facts

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Adding missing SQL Database facts.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
azure_rm_sqldatabase_facts

##### ANSIBLE VERSION
2.7

### CODE REVIEW

**Comment by Fred-sun** [comment]
@zikalino does this PR ready for review, please help confirm? Thanks!

**Comment by Fred-sun** [comment]
@zikalino Could you help to confirm the PR ready for review or not? Thanks!

**Comment by Fred-sun** [comment]
@zikalino Could you help to confirm the PR ready for review or not? Thanks!

**Comment by Fred-sun** [comment]
@zikalino Could you help confirm that do you have finished this PR change? Thanks!

**Comment by zikalino** [comment]
@Fred-sun I will remove [WIP] when it's ready for reviw

**Comment by Fred-sun** [comment]
Thanks @zikalino , I will pay attention on this.

File: lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+            - The name of the database.
+    elastic_pool_name:
+        description:
+            - The name of the elastic pool.
+    tags:
+        description:
+            - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+
+extends_documentation_fragment:
+    - azure
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py:44]: azure_tags?

File: lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                        - The name of the SKU. Ex - P3. It is typically a letter+number code
+                    returned: always
+                    type: str
+                    sample: BC_Gen4_2
+                tier:
+                    description:
+                        - This field is required to be implemented by the Resource Provider if the service has more than one tier, but is not required on a PUT.
+                    returned: always
+                    type: str
+                    sample: BusinessCritical
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py:120]: python way?

File: lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+                - Kind of database. This is metadata used for the Azure portal experience.
+            returned: always
+            type: str
+            sample: v12.0,user
+        collation:
+            description:
+                - The collation of the database.
+            returned: always
+            type: str
+            sample: SQL_Latin1_General_CP1_CI_AS
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py:138]: python way?

File: lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+        status:
+            description:
+                - "The status of the database. Possible values include: 'Online', 'Restoring', 'RecoveryPending', 'Recovering', 'Suspect', 'Offline',
+                   'Standby', 'Shutdown', 'EmergencyMode', 'AutoClosed', 'Copying', 'Creating', 'Inaccessible', 'OfflineSecondary', 'Pausing', 'Paused',
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py]: python way?

File: lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py:44]: that would add documentation fragment supposed for main module. for facts doc is entirely different

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino Please help update according by yungezz's comment? Thanks!

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino Could you help recheck this PR and push for merge when you're available? Thanks!

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino

File: lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+                - Resource location.
+            returned: always
+            type: str
+            sample: southeastasia
+        tags:
+            description:
+                - Resource tags.
+            returned: always
+            type: complex
+            sample: tags
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py]: not proper sample

File: lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.sql import SqlManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMDatabasesFacts(AzureRMModuleBase):
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py]: class name?

File: lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+        self.server_name = None
+        self.name = None
+        self.elastic_pool_name = None
+        self.tags = None
+        super(AzureRMDatabasesFacts, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py]: api version

File: lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+    def list_by_elastic_pool(self):
+        response = None
+        results = []
+        try:
+            response = self.mgmt_client.databases.list_by_elastic_pool(resource_group_name=self.resource_group,
+                                                                       server_name=self.server_name,
+                                                                       elastic_pool_name=self.elastic_pool_name)
+            self.log("Response : {0}".format(response))
+        except CloudError as e:
+            self.log('Could not get facts for Databases.')
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py]: fail?

File: lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+
+    def list_by_server(self):
+        response = None
+        results = []
+        try:
+            response = self.mgmt_client.databases.list_by_server(resource_group_name=self.resource_group,
+                                                                 server_name=self.server_name)
+            self.log("Response : {0}".format(response))
+        except CloudError as e:
+            self.log('Could not get facts for Databases.')
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py]: fail?

File: lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+'''
+
+EXAMPLES = '''
+  - name: Get instance of SQL Database
+    azure_rm_sqldatabase_facts:
+      resource_group: testrg
+      server_name: testserver
+      name: testdb
+
+  - name: List instances of SQL Database
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqldatabase_facts.py:58]: duplicate name with below sample, be more specific

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
shipit

**Pull request merged**